### DISCUSSION TITLE
Disable test_* generation with FetchContent

### DISCUSSION THREAD

**Discussion by mvitorovic** [open-source-parsers/jsoncpp#1580] [Q&A] [answered]
I'm using CMake FetchContent to pull in jsoncpp, but now my Debug output folder also contains `jsoncpp_test` and `jsontestrunner_exe` .  Is there any way to prevent those two files to be generated?

Thanks, Miha

**Comment by amalgamat** [comment, marked as answer]
set options before FetchContent. Example (diasabled everything except static library compilation and strict iso )

option(JSONCPP_WITH_TESTS "Compile and (for jsoncpp_check) run JsonCpp test executables" OFF)
option(JSONCPP_WITH_POST_BUILD_UNITTEST "Automatically run unit-tests as a post build step" OFF)
option(JSONCPP_WITH_WARNING_AS_ERROR "Force compilation to fail if a warning occurs" OFF)
option(JSONCPP_WITH_STRICT_ISO "Issue all the warnings demanded by strict ISO C and ISO C++" ON)
option(JSONCPP_WITH_PKGCONFIG_SUPPORT "Generate and install .pc files" OFF)
option(JSONCPP_WITH_CMAKE_PACKAGE "Generate and install cmake package files" OFF)
option(JSONCPP_WITH_EXAMPLE "Compile JsonCpp example" OFF)
option(JSONCPP_STATIC_WINDOWS_RUNTIME "Use static (MT/MTd) Windows runtime" OFF)

option(BUILD_SHARED_LIBS "Build jsoncpp_lib as a shared library." OFF)
option(BUILD_STATIC_LIBS "Build jsoncpp_lib as a static library." ON)
option(BUILD_OBJECT_LIBS "Build jsoncpp_lib as a object library." OFF)

#using TAG 89e2973 cause newer version tries to use string_view and does not work
FetchContent_Declare(
        jsoncpp_library
        GIT_REPOSITORY https://github.com/open-source-parsers/jsoncpp.git
        GIT_TAG 89e2973
)

FetchContent_MakeAvailable(jsoncpp_library)

**Reply by mvitorovic** [reply]
Thank you for your suggestion.

I copy pasted all your suggested options and I still get:

`-rwxr-xr-x 1 mvitorovic mvitorovic 1806184 Mar 18 11:06 jsoncpp_test`
`-rwxr-xr-x 1 mvitorovic mvitorovic  354736 Mar 18 11:06 jsontestrunner_exe`

I'll keep searching for a solution.

Br, Miha

**Reply by mvitorovic** [reply]
Ahh, yes!

The good old "clear the cache after changing options" issue. Or better even - remove the entire build folder.

Thank you. Works great now!
